config/initializers/deprecation_warnings.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+ActiveSupport::Notifications.subscribe("deprecation.rails") do |_name, _start, _finish, _id, payload|
+  message = payload[:message]
+  callstack = payload[:callstack]
+  deprecation_horizon = payload[:deprecation_horizon]
+  gem_name = payload[:gem_name] || "rails"
+
+  execution_context = ActiveSupport::ExecutionContext.to_h
+  appsignal_action = (execution_context[:controller] || execution_context[:job]).class.to_s
```

Comment:
I guess if these aren't populated we'll get ‘NilClass` which isn't the end of the world